mesonbuild/wrap/wrap.py

```diff
@@ -129,7 +129,9 @@ def __init__(self, fname: str, subproject: str = ''):
         self.redirected = False
         if self.has_wrap:
             self.parse_wrap()
-        self.directory = self.values.get('directory', self.name)
+            self.directory = self.values.get('directory', self.name)
+            with open(fname) as file:
+                self.hash_of_wrap_file = hashlib.sha256(file.read().encode('utf-8')).hexdigest()
```

Comment:
I would use `self.wrapfile_hash`

Reply:
Renamed from `hash_of_wrap_file` to `wrapfile_hash`